### PR TITLE
Allow rot_spawn use monster id directly

### LINKED ISSUE #73858: Allow ``rot_spawn`` to define a monster ID directly instead of needing a monstergroup
### Is your feature request related to a problem? Please describe.

``rot_spawn`` only takes monstergroups. This is evil, because it means for every egg-laying animal that we allow to reproduce we need a custom monstergroup with just one entry for the baby version. This is annoying as hell.

### Solution you would like.

Allow ``rot_spawn`` to take monster IDs directly to cut out the middle man

### Describe alternatives you have considered.

_No response_

### Additional context

It's important to note that it should still be allowed to take monstergroups as well, since some of our eggs do rely on that. I am asking for a new possibility, not to replace the existing one with another.

### LINKED ISSUE #76785: Capability to use item transform actions with itemgroup IDs
### Is your feature request related to a problem? Please describe.

I have recently tried to have an item transformation action for a mod that'd turn into one of multiple targets. This is currently unsupported, and may only be possible (even with Guardian's help we couldn't get it to work without errors) with EoCs.

### Solution you would like.

 It'd be handy to have the item transformation actions support itemgroup IDs directly to save me a headache.

### Describe alternatives you have considered.

Suffer in EoC limbo

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
Close #73858
#### Describe the solution
Slightly redo the way json is read, add few new field
load_min_max() is moved above because i wanted to use it, and didn't want to use forward declaration
json is churned using regex and few manual edits
#### Describe alternatives you've considered
If we have some way to inline monster group defintion, it can be used instead, but i am not aware we have such
#### Testing
Tested new syntax, tested it's inheritance, seems to work same as before
11 calfs born out of 9 eggs with 70% chance to spawn and [ 1 ,3 ] count:
image
#### Additional context
regex used, in order:
```
"rot_spawn": "([A-Za-z0-9_]*)",[\n ]*"rot_spawn_chance": ([0-9]{1,3})
"rot_spawn": { "group": "$1", "chance": $2 }

"rot_spawn": "([A-Za-z0-9_]*)",[\n ]*"rot_spawn_chance": ([0-9]{1,3})
"rot_spawn": { "group": "$2", "chance": $1 }

"rot_spawn_chance": ([0-9]{1,3})
"rot_spawn": { "chance": $1 }

"rot_spawn": "([A-Za-z0-9_]*)"
"rot_spawn": { "group": "$1" }
```

### CODE REVIEW

**Comment by Karol1223** [comment]
Guardian you silly fellow you linked the wrong issue

File: data/json/items/comestibles/egg.json [hunk 1]
```diff
@@ -129,14 +128,15 @@
...
-    "rot_spawn": "GROUP_EGG_DUCK"
+    "rot_spawn": { "group": "GROUP_EGG_DUCK" }
   },
   {
     "type": "COMESTIBLE",
     "id": "egg_duck_domestic",
     "name": { "str": "domestic duck egg" },
     "copy-from": "egg_duck",
-    "rot_spawn": "GROUP_EGG_DUCK_DOMESTIC"
+    "spoils_in": "20 minutes",
```
Comment (Karol1223) [inline data/json/items/comestibles/egg.json:138]:
```suggestion
    "spoils_in": "20 minutes",
```

Nuh huh. That's your testing changes you never reverted.

**Review by Karol1223 — CHANGES_REQUESTED** [review]

File: doc/JSON_INFO.md [hunk 2]
```diff
@@ -4066,8 +4066,13 @@ CBMs can be defined like this:
   { "type": "wheat", "portion": 5 }
 ],
 "primary_material": "meat",       // What the primary material ID is. Materials determine specific heat.
-"rot_spawn": "MONSTERGROUP_NAME", // Monster group that spawns when food becomes rotten (used for egg hatching)
-"rot_spawn_chance": 10,           // Percent chance of monstergroup spawn when food rots. Max 100.
+"rot_spawn": {                    // Defines what creature would be spawned when this item gonna rot. Mostly used for eggs
```
Comment (Karol1223) [inline doc/JSON_INFO.md:4069]:
```suggestion
"rot_spawn": {                    // Defines what creature would be spawned when this item rots away. Primarily used for eggs
```

File: doc/JSON_INFO.md [hunk 3]
```diff
@@ -4066,8 +4066,13 @@ CBMs can be defined like this:
...
+  "group": "GROUP_EGG_CHICKEN",   // id of monster group that would be spawned. Cannot be used with "monster"
+  "monster": "mon_moose_calf",    // id of a monster that would be spawned. Cannot be used with "group"
+  "amount": 2,                    // if "monster" is used, defines how many of this monster will be spawned;
```
Comment (Karol1223) [inline doc/JSON_INFO.md:4072]:
```suggestion
  "amount": 2,                    // if "monster" is used, defines how many instances of this monster will be spawned; 
```

File: doc/JSON_INFO.md [hunk 4]
```diff
@@ -4066,8 +4066,13 @@ CBMs can be defined like this:
...
 ],
 "primary_material": "meat",       // What the primary material ID is. Materials determine specific heat.
-"rot_spawn": "MONSTERGROUP_NAME", // Monster group that spawns when food becomes rotten (used for egg hatching)
-"rot_spawn_chance": 10,           // Percent chance of monstergroup spawn when food rots. Max 100.
+"rot_spawn": {                    // Defines what creature would be spawned when this item gonna rot. Mostly used for eggs
+  "group": "GROUP_EGG_CHICKEN",   // id of monster group that would be spawned. Cannot be used with "monster"
+  "monster": "mon_moose_calf",    // id of a monster that would be spawned. Cannot be used with "group"
+  "amount": 2,                    // if "monster" is used, defines how many of this monster will be spawned; 
+  "amount": [ 1, 3 ],             // also can be an array, then the game will roll the number between two values
+  "chance": 70                    // chance for animal to spawn from single item
```
Comment (Karol1223) [inline doc/JSON_INFO.md:4074]:
```suggestion
  "chance": 70                    // chance for the monster to spawn from a single item
```

**Review by Karol1223 — CHANGES_REQUESTED** [review]

**Comment by Maleclypse** [comment]
conflicts

**Pull request merged**